src/modules/deal/dto/deals-request.dto.ts

```diff
@@ -0,0 +1,73 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsDateString, IsNotEmpty, IsNumber, IsOptional, IsString, MaxLength } from 'class-validator';
+
+export class CreateDealDto {
+  @ApiProperty({
+    example: '0020900035', // Comment on PR if we need a description when it's self evident
+  })
+  @IsString()
+  @IsNotEmpty()
+  @MaxLength(10)
+  dealIdentifier: string;
+
+  @ApiProperty({
+    description: 'Deal portfolio is E1',
+    example: 'E1',
+  })
+  @IsString()
+  @IsOptional()
+  @MaxLength(2)
+  portfolioIdentifier = 'E1';
+
+  @ApiProperty({
+    description: 'Deal currency code is USD, CAD',
+    example: 'GBP',
+  })
+  @IsString()
+  @IsNotEmpty()
+  @MaxLength(3)
+  currency: string;
+
+  @ApiProperty()
+  @ApiProperty({
+    example: 2000000000.9989765,
+  })
+  @IsNumber()
+  @IsNotEmpty()
```

Comment:
Will `IsNotEmpty`  also cover `0.0`?

Reply:
According to the docs, `IsNotEmpty` equates to `!== ''`, `!== null`, `!== undefined`. My intention was to meet the "required" part of the API spec.

What are the requirements for deal values -- should we explicitly forbid a deal value of 0.00? I don't know if there exists a realistic case for this within the domain.